File: materias 3 periodo ple/programacao orientada ao objeto/trabalhos praticos/tp-1/Codigo/Cpp/Pessoa.cpp
#include "../Hpp/Pessoa.hpp"
#include <iostream>

Pessoa :: Pessoa(string nome, string cpf, string rg, string endereco, string telefone, string email) : nome(nome), cpf(cpf), rg(rg), endereco(endereco), telefone(telefone), email(email) {}

Pessoa :: ~Pessoa() {}

// Gets

string Pessoa :: getNome() {
    return nome;
}

string Pessoa :: getCpf() {
    return cpf;
}

string Pessoa :: getRg() {
    return rg;
}

string Pessoa :: getEndereco() {
    return endereco;
}

string Pessoa :: getTelefone() {
    return telefone;
}

string Pessoa :: getEmail() {
    return email;
}

// Sets

void Pessoa :: setNome(string nome) {
    this->nome = nome;
}

void Pessoa :: setCpf(string cpf) {
    this->cpf = cpf;
}

void Pessoa :: setRg(string rg) {
    this->cpf = cpf;
}

void Pessoa :: setEndereco(string endereco) {
    this->endereco = endereco;
}

void Pessoa :: setTelefone(string telefone) {
    this->telefone = telefone;
}

void Pessoa :: setEmail(string email) {
    this->email = email;
}

// Sobrecarga

istream& operator>>(istream& in, Pessoa& pessoa) {
    getchar();
    cout << "Nome: ";
    getline(in, pessoa.nome);
    cout << "CPF: ";
    getline(in, pessoa.cpf);
    cout << "RG: ";
    getline(in, pessoa.rg);
    cout << "Endereço: ";
    getline(in, pessoa.endereco);
    cout << "Telefone: ";
    getline(in, pessoa.telefone);
    cout << "Email: ";
    getline(in, pessoa.email);
    return in;
}

File: materias ple/programacao orientada ao objeto/trabalhos praticos/tp-2/SistemaOficina/src/sistemaoficina/Pessoa/Pessoa.java
package sistemaoficina.Pessoa;

public class Pessoa {
    private String nome;
    private String cpf;
    private String rg;
    private String endereco;
    private String telefone;
    private String email;
    /**
    * Construtor da classe Pessoa
    */
    public Pessoa(String nome, String cpf, String rg, String endereco, String telefone, String email) {
        setNome(nome);
        this.cpf = cpf;
        setRg(rg);
        setEndereco(endereco);
        setTelefone(telefone);
        setEmail(email);
    }

    public Pessoa() {} 

    /**
     * Retorna o valor relacionado ao campo nome
     * @return String nome
     */
    public String getNome() {
        return nome;
    }
    
    /**
     *  Retorna o valor relacionado ao campo cpf
     *  @return String cpf
     */
    public String getCpf() {
        return cpf;
    }
    
    /**
     * Retorna o valor relacionado ao campo rg
     * @return String rg
     */
    public String getRg() {
        return rg;
    } 

    /**
     * Retorna o valor relacionado ao campo endereço
     * @return String endereco
     */
    public String getEndereco() {
        return endereco;
    }

    /**
     * Retorna o valor relacionado ao campo telefone
     * @return String telefone
     */
    public String getTelefone() {
        return telefone;
    }
    
    /**
     * Retorna o valor relacionado ao campo email
     * @return String email
     */
    public String getEmail() {
        return email;
    }
    
    /**
     * Edita o valor da variavel <strong>this</strong> para o valor no parametro
     * @param nome
     */
    protected void setNome(String nome) {
        this.nome = nome;
    }

    /**
     * Edita o valor da variavel <strong>this</strong> para o valor no parametro
     * @param cpf
     */
    protected void setCpf(String cpf) {
        this.cpf = cpf;
    }

    /**
     * Edita o valor da variavel <strong>this</strong> para o valor no parametro
     * @param rg
     */
    protected void setRg(String rg) {
        this.rg = rg;
    }

    /**
     * Edita o valor da variavel <strong>this</strong> para o valor no parametro
     * @param endereco
     */
    protected void setEndereco(String endereco) {
        this.endereco = endereco;
    }

    /**
     * Edita o valor da variavel <strong>this</strong> para o valor no parametro
     * @param telefone
     */
    protected void setTelefone(String telefone) {
        this.telefone = telefone;
    }

    /**
     * Edita o valor da variavel <strong>this</strong> para o valor no parametro
     * @param email
     */
    protected void setEmail(String email) {
        this.email = email;
    }
}